message-bridge/README.adoc

```diff
@@ -0,0 +1,206 @@
+2023-10-13 15:21:46,484 WARN  [com.arj.ats.jta] (Periodic Recovery) ARJUNA016038: No XAResource to recover < formatId=131077, gtrid_length=35, bqual_length=36, tx_uid=0:ffffc0a80136:8e8b:65294417:31, node_name=quarkus, branch_uid=0:ffffc0a80136:8e8b:65294417:33, subordinatenodename=null, eis_name=0 >
+2023-10-13 15:21:46,485 INFO  [org.acm.mes.bri.DummyXAResource] (Periodic Recovery) Committing DummyXAResource
+2023-10-13 15:21:46,512 INFO  [amq] (Camel (camel-1) thread #2 - JmsConsumer[in]) ActiveMQ received: crash
+----
+
+You may observe the warning messages related to the XAResource recovery in the logs. This is a known issue related to how we simulate the crash of the system and can be ignored.
+
+=== Package and run the application
+
+Once you are done with developing you may want to package and run the application.
+
+TIP: Find more details about the JVM mode and Native mode in the Package and run section of
+https://camel.apache.org/camel-quarkus/latest/first-steps.html#_package_and_run_the_application[Camel Quarkus User guide]
+
+==== JVM mode
+
+[source,shell]
+----
+$ mvn clean package
+$ java -jar target/quarkus-app/quarkus-run.jar
+...
+[io.quarkus] (main) camel-quarkus-examples-... started in 1.163s.
+----
+
+=== Running on OpenShift
+
+You can also deploy this example as an OpenShift pod using the capabilities of the quarkus-maven-plugin.
+
+Before doing so, make sure to deploy the AMQ and IBMMQ brokers:
+
+----
+oc new-app quay.io/artemiscloud/activemq-artemis-broker -e AMQ_USER=admin -e AMQ_PASSWORD=admin
+oc patch service/activemq-artemis-broker -p '{"spec":{"ports":[{"name":"61616-tcp", "port": 61616, "protocol": "TCP", "targetPort": 61616}]}}'
+----
+
+----
+oc new-app icr.io/ibm-messaging/mq:9.3.2.1-r1 -e MQ_QMGR_NAME=QM1 -e LICENSE=accept -e MQ_APP_PASSWORD=passw0rd
+----
+
+Next, create a `PersistentVolumeClaim` to serve as the storage for the transaction manager's object store. This example assumes that a persistent volume claim named `message-bridge` has already been set up. Keep in mind that the configuration of the persistent volume may require adjustments based on your specific OpenShift setup:
+
+----
+oc create -f - <<EOF
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: message-bridge
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
+EOF
+----
+
+Then deploy this example using the `openshift` profile specified within this project:
+
+----
+mvn clean package -DskipTests -Popenshift
+----
+
+Once the pod is up and running successfully, you can send messages in a manner similar to local deployment:
+
+----
+curl -X POST -H "Content-Type: text/plain" http://$(oc get route camel-quarkus-examples-message-bridge -o jsonpath='{.spec.host}')/message -d 'Hello world'
+----
+
+=== Running the tests
+
+This example comes with a couple of integration tests that verify the correct functionality. Before running the tests examine the IBM MQ license with:
+
+----
```

Comment:
This one seems not to work, locally it outputs:

```
...
2023-10-16T13:49:24.309Z License not accepted
```

Is it really needed to run this docker container actually.
Indeed, it looks like the `mvn clean verify ...` command runs well without the docker command line executed.

Reply:
technically it is not needed, but to run the ibm mq docker image, you need to accept the license. This only provides a way how to view the license in case you want to read it beforehand.

running the docker container with `-e LICENSE=view` only prints the license and ends with `License not accepted` as you mentioned.

this command comes from the container itself, e.g. if you run just `docker run icr.io/ibm-messaging/mq:9.3.2.1-r1` it ends with:

```
2023-10-17T06:14:37.166Z Error: Set environment variable LICENSE=accept to indicate acceptance of license terms and conditions.
2023-10-17T06:14:37.166Z License agreements and information can be viewed by setting the environment variable LICENSE=view.  You can also set the LANG environment variable to view the license in a different language.
2023-10-17T06:14:37.166Z Error checking license acceptance: Set environment variable LICENSE=accept to indicate acceptance of license terms and conditions
```

to run the tests, you need to use `-Dibm.mq.container.license=accept` and this value is passed to the docker image as the LICENSE env var [here](https://github.com/avano/camel-quarkus-examples/blob/ibmmq-amq/message-bridge/src/test/java/org/acme/message/bridge/resource/IBMMQTestResource.java#L40), otherwise if you don't accept the tests are [skipped](https://github.com/avano/camel-quarkus-examples/blob/ibmmq-amq/message-bridge/src/test/java/org/acme/message/bridge/MessageBridgeITCase.java#L43)

Reply:
Many thanks for sharing, it's understood now :+1: 

Thinking a bit more about the value of integration tests in examples. They actually prevent regressions, as we run the itests at each release when we [upgrade and tag examples](https://camel.apache.org/camel-quarkus/3.4.x/contributor-guide/release-guide.html#_upgrade_and_tag_examples). So, the rationale would be that it's not really up to the example user to read the license but more the developer that commit the code. We have kind of similar mechanism like [that](https://github.com/apache/camel-quarkus/blob/main/integration-test-groups/jdbc/mssql/src/test/resources/container-license-acceptance.txt).

In this case, shouldn't we remove the test section from the README ? And then set the `LICENSE=accept` environment variable in the test config to show that the developers accept the license in the context of itests ?